lib/puppet/provider/ios_aaa_authentication/cisco_ios.rb

```diff
@@ -3,6 +3,15 @@
 
 # Configure AAA Authentication on the device
 class Puppet::Provider::IosAaaAuthentication::CiscoIos
+
+  def canonicalize(_context, resources)
```

Comment:
This will never get called as the type does not have the `canonicalize` feature (example: https://github.com/puppetlabs/cisco_ios/blob/master/lib/puppet/type/ios_config.rb#L6)